decoder/pokemon.go

```diff
@@ -71,6 +74,9 @@ type Pokemon struct {
 	IsEvent                 int8        `db:"is_event"`
 }
 
+const EncounterWeather_Invalid uint8 = 0xFF  // invalid/unscanned
+const EncounterWeather_Rerolled uint8 = 0x80 // flag for marking that the spawn rerolled since last scan
```

Comment:
I followed the existing code style. See `stats_seenWild`. Should we change those too?

Reply:
Yes, please do. I was planning to do cleanup (including those) and did one before, but there are so many changes right now that it makes no sense.

Reply:
Fixed these two variables. I will leave fixing the rest of the variables to you. :)

Reply:
Thanks. I'll fire up the scripts when it's calmer

Reply:
... the naming convention of these consts is to follow the protobuf generated code

Reply:
(I'm not saying that we should necessarily do that, but that is why the object_name model was chosen)

Reply:
> ... the naming convention of these consts is to follow the protobuf generated code

...and is not within protobuf but project itself so should follow language rules. You are the owner, I'm backing off.

Reply:
The right thing to do is probably to create a new type, and then the constants are collected together.
The problem with consts that are not discriminated is that you otherwise can't tell how they are grouped or that they are options within the same category.
This naming solves that, but in absence of that then a type overriding uint8 (or whatever) should be used

Reply:
Let me think on it.
The protobuf naming convention is because there isn't namespacing inside a module in golang. And it's a valid concern because without a consistent prefix to group these constants together it is a free for all.

https://go.dev/play/p/JpmKvLA7WLG
this (above) would be how things are grouped in the protobuf generated files, as an example - and is the naming convention that I followed in the original code to group all constants together that were related.

Reply:
I changed back for merging for now. Once you decide you can refactor. :)